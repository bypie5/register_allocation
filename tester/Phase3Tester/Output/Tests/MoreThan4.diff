--- SelfTestCases/MoreThan4.out	2011-04-28 14:55:20.000000000 -0700
+++ ./Output/Tests/MoreThan4.out	2020-02-19 16:01:55.000000000 -0800
@@ -1,13 +1 @@
-1
-2
-3
-4
-5
-6
-6
-5
-4
-3
-2
-1
-0
+Program has no "Main" function.
