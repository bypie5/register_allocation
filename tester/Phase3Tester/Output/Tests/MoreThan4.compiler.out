const vmt_MT4
    :MT4.Start
    :MT4.Change

func Main [in 0, out 3, local 3]
local[0] = $s0
local[1] = $s1
local[2] = $s2
$s0 = HeapAllocZ(4)
[$s0+0] = :vmt_MT4
if $s0 goto :null1
Error("null pointer")
null1:
$s1 = [$s0+0]
$s1 = [$s1+0]
$a0 = $s0
$a1 = 1
$a2 = 2
$a3 = 3
out[0] = 4
out[1] = 5
out[2] = 6
call $s1
$s2 = $v0
PrintIntS($s2)
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
ret

func MT4.Start [in 3, out 3, local 8]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
local[4] = $s4
local[5] = $s5
local[6] = $s6
local[7] = $s7
$s3 = $a0
$s0 = $a1
$s4 = $a2
$s1 = $a3
$s5 = in[0]
$s2 = in[1]
$s6 = in[2]
PrintIntS($s0)
PrintIntS($s4)
PrintIntS($s1)
PrintIntS($s5)
PrintIntS($s2)
PrintIntS($s6)
$s7 = [$s3+0]
$s7 = [$s7+4]
$a0 = $s3
$a1 = $s6
$a2 = $s2
$a3 = $s5
out[0] = $s1
out[1] = $s4
out[2] = $s0
call $s7
$t0 = $v0
$v0 = $t0
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
$s4 = local[4]
$s5 = local[5]
$s6 = local[6]
$s7 = local[7]
ret

func MT4.Change [in 3, out 0, local 7]
local[0] = $s0
local[1] = $s1
local[2] = $s2
local[3] = $s3
local[4] = $s4
local[5] = $s5
local[6] = $s6
$s0 = $a0
$s1 = $a1
$s2 = $a2
$s3 = $a3
$s4 = in[0]
$s5 = in[1]
$s6 = in[2]
PrintIntS($s1)
PrintIntS($s2)
PrintIntS($s3)
PrintIntS($s4)
PrintIntS($s5)
PrintIntS($s6)
$v0 = 0
$s0 = local[0]
$s1 = local[1]
$s2 = local[2]
$s3 = local[3]
$s4 = local[4]
$s5 = local[5]
$s6 = local[6]
ret

